### PR TITLE
docs(samples): add samples to create/delete featurestore

### PR DESCRIPTION
**Pull request by nayaknishant**
Migrating GAPIC feature store samples to SDK. Completed 2/8 samples with units tests, would appreciate feedback.

### CODE REVIEW

File: samples/model-builder/delete_featurestore_sample_test.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Google LLC
```
Comment (morgandu) [inline samples/model-builder/delete_featurestore_sample_test.py:1]: 2022

File: samples/model-builder/test_constants.py [hunk 2]
```diff
@@ -103,6 +104,7 @@
 MACHINE_TYPE = "n1-standard-4"
 ACCELERATOR_TYPE = "ACCELERATOR_TYPE_UNSPECIFIED"
 ACCELERATOR_COUNT = 0
+ONLINE_STORE_FIXED_NODE_COUNT = 1
```
Comment (morgandu) [inline samples/model-builder/test_constants.py:107]: Create a section for feature store constants

**Review by morgandu — CHANGES_REQUESTED** [review]

File: samples/model-builder/conftest.py [hunk 3]
```diff
@@ -364,3 +364,37 @@ def mock_endpoint_explain(mock_endpoint):
...
+"""
+----------------------------------------------------------------------------
+FeatureStore Fixtures
+----------------------------------------------------------------------------
+"""
+
+
+@pytest.fixture
+def mock_featurestore():
+    mock = MagicMock(aiplatform.Featurestore)
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/conftest.py]: Fixed

File: samples/model-builder/conftest.py [hunk 4]
```diff
@@ -364,3 +364,37 @@ def mock_endpoint_explain(mock_endpoint):
...
+    yield mock
+
+
+@pytest.fixture
+def mock_init_featurestore(mock_featurestore):
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/conftest.py]: Fixed

File: samples/model-builder/conftest.py [hunk 5]
```diff
@@ -364,3 +364,37 @@ def mock_endpoint_explain(mock_endpoint):
...
+def mock_create_featurestore(mock_featurestore):
+    with patch.object(aiplatform.Featurestore, "create") as mock_create_featurestore:
+        mock_create_featurestore.return_value = mock_featurestore
+        yield mock_create_featurestore
+
+
+@pytest.fixture
+def mock_delete_featurestore(mock_featurestore):
+    with patch.object(mock_featurestore, "delete") as mock_delete_featurestore:
+        mock_delete_featurestore.return_value = mock_featurestore
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/conftest.py]: Removed setting return value line

File: samples/model-builder/create_featurestore_sample.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    featurestore_id: str,
+    online_store_fixed_node_count: int = 1,
+):
+
+    aiplatform.init(project=project, location=location)
+
+    fs = aiplatform.Featurestore.create(
+        featurestore_id=featurestore_id,
+        online_store_fixed_node_count=online_store_fixed_node_count,
+    )
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/create_featurestore_sample.py:34]: Done

File: samples/model-builder/create_featurestore_sample_test.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
+# Copyright 2021 Google LLC
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/create_featurestore_sample_test.py]: Fixed

File: samples/model-builder/create_featurestore_sample_test.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        featurestore_id=constants.FEAUTURESTORE_NAME,
+        online_store_fixed_node_count=constants.ONLINE_STORE_FIXED_NODE_COUNT,
+    )
+
+    mock_sdk_init.assert_called_once_with(
+        project=constants.PROJECT, location=constants.LOCATION
+    )
+    mock_create_featurestore.assert_called_once_with(
+        featurestore_id=constants.FEAUTURESTORE_NAME,
+        online_store_fixed_node_count=constants.ONLINE_STORE_FIXED_NODE_COUNT,
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/create_featurestore_sample_test.py:35]: Added `sync` arg

File: samples/model-builder/delete_featurestore_sample.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+def delete_featurestore_sample(
+    project: str, location: str, featurestore_name: str,
+):
+
+    aiplatform.init(project=project, location=location)
+
+    fs = aiplatform.Featurestore(featurestore_name=featurestore_name)
+    fs.delete()
+
+    return fs
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/delete_featurestore_sample.py]: Fixed

File: samples/model-builder/delete_featurestore_sample_test.py (hunk 1 above)
Comment (nayaknishant) [reply samples/model-builder/delete_featurestore_sample_test.py]: Fixed

File: samples/model-builder/delete_featurestore_sample_test.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+    mock_sdk_init.assert_called_once_with(
+        project=constants.PROJECT, location=constants.LOCATION
+    )
+
+    mock_init_featurestore.assert_called_once_with(
+        featurestore_name=constants.FEAUTURESTORE_NAME
+    )
+
+    mock_delete_featurestore.assert_called_once_with()
```
[the remark this answers is not in the document]
Comment (nayaknishant) [reply samples/model-builder/delete_featurestore_sample_test.py]: Done

File: samples/model-builder/test_constants.py (hunk 2 above)
Comment (nayaknishant) [reply samples/model-builder/test_constants.py]: Sounds good

File: samples/model-builder/create_featurestore_sample_test.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+        sync=constants.SYNC,
+    )
+
+    mock_sdk_init.assert_called_once_with(
+        project=constants.PROJECT, location=constants.LOCATION
+    )
+
+    mock_create_featurestore.assert_called_once_with(
+        featurestore_id=constants.FEAUTURESTORE_NAME,
```
Comment (morgandu) [inline samples/model-builder/create_featurestore_sample_test.py:34]:
```suggestion
        featurestore_id=constants.FEAUTURESTORE_ID,
```

Comment (morgandu) [reply samples/model-builder/create_featurestore_sample_test.py:34]: See docstring for detail https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform/featurestore/featurestore.py#L413

here `featurestore_name` aka fully qualified resource name can not be applied, only `featurestore_id` aka ID of the featurestore

File: samples/model-builder/test_constants.py [hunk 12]
```diff
@@ -197,3 +197,9 @@
 )
 PYTHON_MODULE_NAME = "trainer.task"
 MODEL_TYPE = "CLOUD"
+
+# Feature store constants
+FEAUTURESTORE_NAME = "featurestore_sample"
```
Comment (morgandu) [inline samples/model-builder/test_constants.py:202]: For class init, the arg `featurestore_name` can take both fully qualified resource name, as well as a resource ID. For create method, the `featurestore_id` takes only ID, not the fully qualified resource name. 

Prefer to have two constants, `FEAUTURESTORE_NAME` and `FEAUTURESTORE_ID` and use them in corresponding settings.

Comment (morgandu) [inline samples/model-builder/test_constants.py:202]:
```suggestion
FEAUTURESTORE_NAME = "projects/123/locations/us-central1/featurestores/featurestore_id"
FEAUTURESTORE_ID = "featurestore_id"
```

File: samples/model-builder/create_featurestore_sample_test.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+import create_featurestore_sample
+import test_constants as constants
+
+
+def test_create_featurestore_sample(mock_sdk_init, mock_create_featurestore):
+
+    create_featurestore_sample.create_featurestore_sample(
+        project=constants.PROJECT,
+        location=constants.LOCATION,
+        featurestore_id=constants.FEAUTURESTORE_NAME,
```
Comment (morgandu) [inline samples/model-builder/create_featurestore_sample_test.py:24]:
```suggestion
        featurestore_id=constants.FEAUTURESTORE_ID,
```

**Review by morgandu — CHANGES_REQUESTED** [review]

File: samples/model-builder/test_constants.py (hunk 12 above)
Comment (nayaknishant) [reply samples/model-builder/test_constants.py]: yep, sounds good

**Review by morgandu — APPROVED** [review]

**Comment by morgandu** [comment]
LGTM! Thanks!

**Pull request merged**